qa/testing.go

```diff
@@ -256,16 +256,17 @@ func (f ResourceFixture) requiresNew(diff *terraform.InstanceDiff) error {
 	return nil
 }
 
-// ApplyNoError is a convenience method for no-data tests
-func (f ResourceFixture) ApplyNoError(t *testing.T) {
-	_, err := f.Apply(t)
-	assert.NoError(t, err, err)
+// ApplyNoError is a convenience method that asserts no error occurred.
+func (f ResourceFixture) ApplyNoError(t *testing.T) *schema.ResourceData {
```

Comment:
Can we not add the `*schema.ResourceData` as the return type?.. we have `ApplyAndExpectData` to assert for data in a more readable way.

The main intent of `ApplyNoError` is to make tests more readable. I want to prevent unreadable tests proliferate with copy-paste

Reply:
* `Apply` returns `d, err`
* `ApplyNoError` returns `d`
* `ApplyAndExpectData` returns nothing

Seems reasonable to me.

What about this approach is not / less readable?

Reply:
I reverted back to the original intent and approach of this PR; only replace `Error(t, err, err)` and `NoError(t, err, err)`.